modules/api/common_types.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2022 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Service contains the fields that are needed for each OpenStack service CRD
+type Service struct {
```

Comment:
a service is a standard type in kubernetes so we should not use this name

also we should likely embed this struct in the other operator rather then having it as a subfield via composition.

i would call it CRDBase

Reply:
Good point about the naming. Let's call it OpenStackServiceBase. It is not CRDBase as there are CRDs which will not use it directly like the Nova CRD.

Could you be more specific what do you mean by embed vs. composition? In the commit message I suggest composition in a way that inlines the struct directly as fields on the API.

Reply:
ya ok i cna get behind OpenStackServiceBase as the name your right its not the base of all CRDs

Reply:
Done. I also refreshed the placement pull request using this in https://github.com/openstack-k8s-operators/placement-operator/pull/29